arrows/vxl/hashed_image_classifier_filter.cxx

```diff
@@ -0,0 +1,168 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+#include "hashed_image_classifier.h"
+#include "hashed_image_classifier_filter.h"
+
+#include <arrows/vxl/image_container.h>
+
+#include <vital/vital_config.h>
+
+#include <vil/algo/vil_threshold.h>
+#include <vil/vil_convert.h>
+#include <vil/vil_image_view.h>
+#include <vil/vil_math.h>
+#include <vil/vil_plane.h>
+
+#include <cstdlib>
+#include <limits>
+#include <type_traits>
+
+namespace kwiver {
+
+namespace arrows {
+
+namespace vxl {
+
+// ----------------------------------------------------------------------------
+/// Private implementation class
+class hashed_image_classifier_filter::priv
+{
+public:
+  // --------------------------------------------------------------------------
+  priv( hashed_image_classifier_filter* parent ) : p{ parent }
+  {
+  }
+
+  // Convert the type
+  template < typename ipix_t > vil_image_view< ipix_t >
+  convert( vil_image_view_base_sptr& view );
+
+  // Scale and convert the image
+  bool
+  load_model();
+
+  hashed_image_classifier_filter* p;
+
+  hashed_image_classifier< vxl_byte, double > hashed_classifier;
+  bool model_loaded{ false };
+
+  std::string model_file{ "" };
+};
+
+// ----------------------------------------------------------------------------
+bool
+hashed_image_classifier_filter::priv
+::load_model()
+{
+  if( !model_loaded )
+  {
+    if( !hashed_classifier.load_from_file( model_file ) )
+    {
+      LOG_ERROR( p->logger(),
+                 "Could not load model_file model" );
+      return false;
+    }
+    model_loaded = true;
+  }
+  return true;
+}
+
+// ----------------------------------------------------------------------------
+hashed_image_classifier_filter
+::hashed_image_classifier_filter()
+  : d( new priv( this ) )
+{
+  attach_logger( "arrows.vxl.hashed_image_classifier_filter" );
+}
+
+// ----------------------------------------------------------------------------
+hashed_image_classifier_filter
+::~hashed_image_classifier_filter()
+{
+}
+
+// ----------------------------------------------------------------------------
+vital::config_block_sptr
+hashed_image_classifier_filter
+::get_configuration() const
+{
+  // get base config from base class
+  vital::config_block_sptr config = algorithm::get_configuration();
+
+  config->set_value( "model_file", d->model_file,
+                     "model_file to load weights from." );
```

Comment:
```suggestion
                     "Model file from which to load weights." );
```
(Richard Dean Anderson [is not amused](https://youtu.be/50OXJ5AT3ms?t=9) :wink:)